packages/agents/src/agents/Chameleon/index.ts

```diff
@@ -159,9 +138,14 @@ export class ChameleonAgent extends NewAgent<GoalRunArgs> {
         IMPORTANT: Respond only with the new text!`
       ).toString();
 
-    const filteredText = await this.askLlm(prompt, charsToTokens(maxCharsToUse - prompt.length));
 
-    console.log("filteredText", text);
+    const filteredText = await this.askLlm(prompt, Math.floor(charsToTokens(maxCharsToUse - prompt.length)));
+
+    console.log("filteredText", filteredText);
+    console.log("maxCharsToUse", maxCharsToUse);
+    console.log("charsForPreview", charsForPreview);
+    console.log("maxCharsToUse - prompt.length", maxCharsToUse - prompt.length);
+    console.log("filteredText.length", filteredText.length);
```

Comment:
Are these logs needed?

Reply:
Yes, they really help with analyzing the behavior :)